validator/src/main.rs

```diff
@@ -891,7 +898,33 @@ fn rpc_bootstrap(
                                 snapshot_hash,
                                 use_progress_bar,
                                 maximum_snapshots_to_retain,
+                                &Some(|download_progress: &DownloadProgressRecord| {
+                                    debug!("Download progress: {:?}", download_progress);
+
+                                    if download_progress.last_throughput <  minimal_snapshot_download_speed
+                                       && download_progress.notification_count <= 1
+                                       && download_progress.percentage_done <= 2_f32
+                                       && download_progress.estimated_remaining_time > 60_f32
+                                       && download_abort_count < maximum_snapshot_download_abort {
+                                        if let Some(ref trusted_validators) = validator_config.trusted_validators {
+                                            if trusted_validators.contains(&rpc_contact_info.id) && trusted_validators.len() == 1 {
```

Comment:
I think there's a missing `no_untrusted_rpc` check here.  It's ok to abort there's only a single trusted validator and the user hasn't restricted RPC to trusted validators only

Reply:
Good point. Will address

Reply:
Fixed